### PR TITLE
Add PDFDrive Search Functionality!

### PR DESCRIPTION
**Pull request by wearrrrr**
**What kind of change does this PR introduce?**

Add PDFDrive as a way to search for books

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
no

**Summary**

It was an issue on github and I at least wanted to get the ball rolling a little bit.

### CODE REVIEW

File: src/models/types.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import { Book, Hashes } from './base-types';
+import PdfDrive from '../providers/books/pdfdrive';
+import { type } from 'os';
```
Comment (riimuru) [inline src/models/types.ts:3]: `PdfDrive` & `type` are imported but never used

File: src/providers/books/pdfdrive.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import axios from 'axios';
+import { load } from 'cheerio';
+import BookParser from '../../models/book-parser';
+import { ISearch, IBookResult } from '../../models/types';
+
+class PdfDrive extends BookParser {
+    protected override readonly baseUrl = 'https://pdfdrive.com';
+    /**
+    * @type {string}
```
Comment (riimuru) [inline src/providers/books/pdfdrive.ts:9]: unnecessary jsdoc comment

File: src/providers/books/pdfdrive.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    */
+    override readonly name: string = 'PdfDrive';
+    protected override logo = "https://www.pdfdrive.com/assets/img/logo-og-2.png"
+    protected override classPath = "Books.PdfDrive";
+    
+    override search = async (query: string, page: number = 1, pageCount: string = "", publishedYear: string = "", bookLanguage: string = "", exactMatch: boolean = false): Promise<ISearch<IBookResult>> => {
```
Comment (riimuru) [inline src/providers/books/pdfdrive.ts:15]: better to use `string?` instead of `string = ""`

**Review by riimuru — CHANGES_REQUESTED** [review]

File: src/models/types.ts (hunk 1 above)
Comment (wearrrrr) [reply src/models/types.ts:3]: Oops, thought I removed these, I'll fix all of them when im home

File: src/providers/books/pdfdrive.ts (hunk 3 above)
Comment (wearrrrr) [reply src/providers/books/pdfdrive.ts:15]: There will *always* be this information from the book afaik, but at least title should be required

Comment (wearrrrr) [reply src/providers/books/pdfdrive.ts:15]: And also I used ? later in the function so you still only have to pass in 1 or 2 args for it to work

**Pull request closed without merging**